README.md

```diff
@@ -1,94 +1,125 @@
-
-There are a couple of really simple test services -- [win penguin](https://github.com/buildkite/agent-stack-k8s) and [fail whale](https://github.com/buildkite/agent-stack-k8s). We also pushed the images for [win penguin](https://hub.docker.com/repository/docker/deftinc/winpenguin) and [fail whale](https://hub.docker.com/repository/docker/deftinc/failwhale) up to Docker Hub for testing.
-
-## Running the stack locally
-
-The easiest way to get started is to run a kind (kubernetes-in-docker) cluster on your local machine. We have a few scripts that make cluster provisioning and bootstrap easier.
-
-You'll need to have the docker community edition installed to run the local tests with kind (kubernetes-in-docker). We recommend installing it directly from [https://www.docker.com/get-started/](https://www.docker.com/get-started/)
-
-The rest of the local development environment dependencies are managed with [homebrew](https://brew.sh) and can be installed with the bootstrap script:
-
-```
-./bin/bootstrap
+## Architecture
+
+```mermaid
+sequenceDiagram
+    participant bc as buildkite controller
+    participant gql as Buildkite GraphQL API
+    participant bapi as Buildkite API
+    participant kubernetes
+    bc->>gql: Get scheduled builds & jobs
+    gql-->>bc: {build: jobs: [{uuid: "abc"}]}
+    kubernetes->>pod: start
+    bc->>kubernetes: watch for pod completions
+    bc->>kubernetes: create pod with agent sidecar
+    kubernetes->>pod: create
+    pod->>bapi: agent accepts & starts job
+    pod->>pod: agent bootstrap
+    pod->>pod: run user pod to completion
+    pod->>bapi: upload artifacts, exit code
+    pod->>pod: exit
+    kubernetes->>bc: pod completion event
+    bc->>kubernetes: cleanup finished pods
 ```
 
-## Running the stack locally
-
-The easiest way to get started is to run a kind(kubernetes-in-docker) cluster on your local machine. We have a few scripts that make cluster provisioning and bootstrap easier.
-
-To get started run the command below. This will setup a single node kubernetes cluster in docker, add the kubernetes metric server, and then add manifests for the buildkite kubernetes stack.
-
-```
-./bin/up k8s/overlay/my-stackname
+## Example execution pod
+
+```yaml!
+apiVersion: v1
+kind: Pod
+metadata:
+  name: agent-abc
+spec:
+  initContainers:
+  - image: buildkite/agent:latest
+    args:
+    - start
+    - --acquire-job=abc
+    volumeMounts:
```

Comment:
I know this is a future thing, but we might want to say something early on in the readme about “workspaces” and checking out the code into these volumes

Reply:
Added more description of the workspace in the "Explain it like I'm 5" section